### PR TITLE
Allow Magma feedstocks to use GPU runners

### PR DESCRIPTION
**Pull request by carterbox**
## Guidelines for marking packages as broken:

* We prefer to patch the repo data (see [here](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock))
  instead of marking packages as broken. This alternative workflow makes environments more reproducible.
* Packages with requirements/metadata that are too strict but otherwise work are
  not technically broken and should not be marked as such.
* Packages with missing metadata can be marked as broken on a temporary basis
  but should be patched in the repo data and be marked unbroken later.
* In some cases where the number of users of a package is small or it is used by
  the maintainers only, we can allow packages to be marked broken more liberally.
* We (`conda-forge/core`) try to make a decision on these requests within 24 hours.

What will happen when a package is marked broken?

* Our bots will add the `broken` label to the package. The `main` label will remain on the package and this is normal.
* Our bots will rebuild our repodata patches to remove this package from the repodata.
* In a few hours after the `anaconda.org` CDN picks up the new patches, you will no longer be able to install the package from the `main` channel.

## Checklist:

* [ ] I want to mark a package as broken (or not broken):
  * [ ] Make sure your package is in the right spot (`broken/*` for adding the
    `broken` label, `not_broken/*` for removing the `broken` label, or `token_reset/*`
    for token resets)
  * [ ] Added a description of the problem with the package in the PR description.
  * [ ] Pinged the team for the package for their input.

* [ ] I want to archive a feedstock:
  * [ ] Pinged the team for that feedstock for their input.
  * [ ] Make sure you have opened an issue on the feedstock explaining why it was archived.
  * [ ] Linked that issue in this PR description.
  * [ ] Added links to any other relevant issues/PRs in the PR description.

* [x] I want to build a package on GPU runners

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @jaimergp @isuruf

**Comment by carterbox** [comment]
https://github.com/Quansight/open-gpu-server/pull/16

File: grant_access/gpu-runner-pr/magma.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+libmagma
```
Comment (isuruf) [inline grant_access/gpu-runner-pr/magma.txt:1]: Just checking. Do you want to build PRs too?

Comment (carterbox) [reply grant_access/gpu-runner-pr/magma.txt:1]: Yes, please. Why not?

File: grant_access/gpu-runner/magma.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+libmagma
```
Comment (isuruf) [inline grant_access/gpu-runner/magma.txt:1]: You don't need both these files

**Comment by carterbox** [comment]
It was unclear to me from the README that gpu-runner-pr implies gpu-runner. i.e. That the permissions granted to a feedstock in gpu-runner are a subset of those granted to a feedstock in gpu-runner-pr.

**Comment by jaimergp** [comment]
Thanks @carterbox! I'll make sure to document this nicely.

**Comment by jakirkham** [comment]
Maybe this is too wordy. Perhaps this could be

* `gpu-runner-branches-only`
* `gpu-runner-branches-and-prs`

Or some other more succinct option

(assuming I even understood the intent correctly 😅)

**Comment by jaimergp** [comment]
Added https://github.com/conda-forge/admin-requests/pull/886
